In [18]:
import pandas as pd
import numpy as np

In [19]:
# load in (cleaned) table containing race info
df = pd.read_csv("/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/cleaned_small/pcrpatientracegroup.csv")

In [23]:
df.head()

PcrPatientRaceGroupKey    PcrKey  ePatient_14
0                 2088217   1472192      7701003
1                16835623  13273412      2514011
2                18418102  14548133      7701003
3                21377926  17184298      2514011
4                38817731  21699657      2514011

In [24]:
unique_values = list(df.ePatient_14.unique())
print(unique_values)

[7701003, 2514011, 2514001, 2514005, 2514007, 7701001, 2514009, 2514003]


- 2514001 - American Indian or Alaska Native 
- 2514003 - Asian
- 2514005 - Black or African American 
- 2514007 - Hispanic or Latino
- 2514009 - Native Hawaiian or Other Pacific Islander
- 2514011 - White
- 7701001 - Not Applicable
- 7701003 - Not Recorded

In [25]:
len(df.index)

49555288

- total observations = 49,555,288
- According to https://nemsis.org/using-ems-data/request-research-data/ the dataset includes 48,982,990 EMS activations, thus 572,298 patients in the dataset are listed as multiple races (this is perfectly OK)

In [26]:
# check race proportions
prop = df['ePatient_14'].value_counts(normalize=True)
prop

2514011    0.432621
7701003    0.301635
2514005    0.156769
2514007    0.065644
7701001    0.023590
2514003    0.010136
2514001    0.006942
2514009    0.002662
Name: ePatient_14, dtype: float64

- I'm gonna drop all of the entries with missing race data
- 30% not recorded, 2.3% not applicable
    - this is sorta interesting, not recorded is obviously just missing data but not applicable could be people that don't fit into any of these categories?


In [27]:
# drop not recorded and n/a
# pandas uses | as the symbol for "or"
nan_obs = df[ (df['ePatient_14'] == 7701003) | (df['ePatient_14'] == 7701001)].index
df.drop(nan_obs, inplace = True)
df.head()

PcrPatientRaceGroupKey    PcrKey  ePatient_14
1                 16835623  13273412      2514011
3                 21377926  17184298      2514011
4                 38817731  21699657      2514011
6                 97354357  72057010      2514011
11               111683931  83733589      2514001

In [28]:
len(df.index)

33438658

In [29]:
# check new proportions
prop = df['ePatient_14'].value_counts(normalize=True)
prop

2514011    0.641134
2514005    0.232328
2514007    0.097283
2514003    0.015021
2514001    0.010288
2514009    0.003946
Name: ePatient_14, dtype: float64

- 64% white
- 23% black
- 10% hispanic / latino
- 2% asian